src/DirectoryCleaner.php

```diff
@@ -27,24 +30,42 @@ public function setDirectory(string $directory)
         return $this;
     }
 
-    public function deleteFilesOlderThanMinutes(int $minutes) : Collection
+    public function setMinutes($minutes)
     {
-        $timeInPast = Carbon::now()->subMinutes($minutes);
+        $this->timeInPast = Carbon::now()->subMinutes($minutes);
+
+        return $this;
+    }
 
-        return collect($this->filesystem->allFiles($this->directory, true))
-            ->filter(function ($file) use ($timeInPast) {
+    public function deleteFilesOlderThanMinutes($amountOfFilesDeleted = 0, $directory = ''): int
+    {
+        if ($directory === '') { // Empty directory given - lets take the default folder, and all sub directories.
+            $directory = realpath($this->directory);
+            $directories = collect(array_merge($this->filesystem->directories($directory), [$directory]));
+        } else {
```

Comment:
Could you try to avoid this `else` statement?

Reply:
Done that. Thanks for pointing out.